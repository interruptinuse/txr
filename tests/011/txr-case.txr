@(load `@{stdlib}txr-case`)
@(define date (year month day))
@{year /\d\d\d\d/}-@{month /\d\d/}-@{day /\d\d/}
@(end)
@(do
   (each ((date '("09-10-20" "2009-10-20" "July-15-2014" "foo")))
     (txr-if date (y m d) date
       (put-line `match: year @y, month @m, day @d`)
       (put-line `no match for @date`))))
@(define notmatch ())
blah
@(end)
@(define stuff (year month day a b))
@(date year month day)
@a @b
@(end)
@(do (txr-case (make-strlist-input-stream '("2021-06-16"
                                            "foo bar"))

       (notmatch () (put-line "notexpected"))
       (stuff (y m d a b) (put-line `match: year @y, month @m, day @d, @a:@b`))))
