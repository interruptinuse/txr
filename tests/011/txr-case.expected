no match for 09-10-20
match: year 2009, month 10, day 20
no match for July-15-2014
no match for foo
match: year 2021, month 06, day 16, foo:bar
